docs/architecture.md

```diff
@@ -0,0 +1,155 @@
+# browsertrix-harvester architecture
+
+This CLI application extends [Browsertrix-Crawler](https://github.com/webrecorder/browsertrix-crawler) as the base docker image, providing an Ubuntu container that has both the somewhat complex browsertrix-crawler installed and configured, and a Pipenv virtual environment for the CLI app that is exposed for use.
+
+```dockerfile
+# extend the browsertrix-crawler docker image
+FROM webrecorder/browsertrix-crawler:latest
+# ...
+# ...
+```
+
+NOTE: this is different from other python CLI apps, which generally use `python:3.11-slim` as the base image.
+
+## Web Crawls
+
+Any actions that trigger a browsertrix web crawl will not work outside of a container context.  A decorator `harvester.utils.require_container` has been created that can be used to decorate functions or methods that should not run outside of a container context.  This decorator looks for EITHER of the following conditions to be true:
+  * the file `/.dockerenv` exists; indicates locally running container
+  * the env var `AWS_EXECUTION_ENV` is set; indicates Fargate ECS task
+
+At this time, only the method `Crawler.crawl` has this treatment.
+
+## Code Architecture
+
+```mermaid
+---
+title: Class Diagrams
+---
+classDiagram
+    class Cli{
+        harvest()        
+    }
+    class Crawler{
+        crawl_name: str
+        config_yaml_filepath: str
+        sitemap_from_date: str
+        btrix_args_json: str[JSON]
+        @wacz_filepath: str
+        @crawl_output_dir: str
+        crawl() -> WACZ archive
+    }
+    class CrawlParser{
+        wacz_filepath: str
+        @archive: ZipFile
```

Comment:
What does the `@` symbol mean? 🤔

Reply:
This is meant to convey that `archive` is a _property_ on the class, and not a directly set value.  But I admit that's probably not obvious.

It felt more accurate than `archive()` which I would interpret to be a callable method, as a user of this class can treat it like a value/property.  

But happy to remove if we don't like this convention.